src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_BuildSynchronization.cs

```diff
@@ -0,0 +1,194 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Globalization;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Diagnostics.EngineV2
+{
+    internal partial class DiagnosticIncrementalAnalyzer : BaseDiagnosticIncrementalAnalyzer
+    {
+        public override async Task SynchronizeWithBuildAsync(Workspace workspace, ImmutableDictionary<ProjectId, ImmutableArray<DiagnosticData>> map)
+        {
+            if (!PreferBuildErrors(workspace))
+            {
+                // prefer live errors over build errors
+                return;
+            }
+
+            var solution = workspace.CurrentSolution;
+            foreach (var projectEntry in map)
+            {
+                var project = solution.GetProject(projectEntry.Key);
+                if (project == null)
+                {
+                    continue;
+                }
+
+                // REVIEW: is build diagnostic contains suppressed diagnostics?
```

Comment:
Build will never report suppressed diagnostics - at least it is not supported right now. If /errorlog is specified to build, the error log will have suppressed diagnostics, but command line output will not.